crates/common/src/backup/manager.rs

```diff
@@ -0,0 +1,382 @@
+        self.l1_processing_lock.lock().await
+    }
+
+    /// Acquires a lock for L2 block processing.
+    /// Should be held while any L2 block operation is taken
+    pub async fn start_l2_processing(&self) -> MutexGuard<'_, ()> {
+        self.l2_processing_lock.lock().await
+    }
+
+    /// Register a database with the backup manager.
+    ///
+    /// # Arguments
+    /// * `path` - Identifier/path for the database (e.g. "ledger", "state", "native-db")
+    /// * `db` - Reference to the database instance
+    ///
+    /// # Returns
+    /// * `Ok(())` if registration succeeds
+    /// * `Err` if the path is not in the configured backup directories
+    ///
+    /// # Errors
+    /// Returns error if the provided path is not present in backup_dirs configuration
+    pub fn register_database(
+        &self,
+        path: String,
+        db: Arc<sov_schema_db::DB>,
+    ) -> anyhow::Result<()> {
+        ensure!(
+            self.config.backup_dirs.contains(&path),
+            "Unexpect database identifier"
+        );
+        self.databases.write().unwrap().insert(path, db);
+        Ok(())
+    }
+
+    /// Creates a backup of all the databases held in config at the specified path.
+    ///
+    /// Acquires both L1 and L2 processing locks to ensure consistency between dbs
+    ///
+    /// # Arguments
+    /// * `path`      - Base directory where the backup will be created
+    /// * `l2_height` - L2 height at which the backup was created
+    ///
+    /// # Returns
+    /// Information about the created backup including block height, path and timestamp
+    pub(super) async fn create_backup(
+        &self,
+        path: Option<PathBuf>,
+        l2_height: u64,
+    ) -> anyhow::Result<CreateBackupInfo> {
+        let backup_path = path
+            .as_ref()
+            .or(self.base_path.as_ref())
+            .context("Missing path and no backup_path found in config.")?;
+
+        let l1_lock = self.l1_processing_lock.lock().await;
+        let l2_lock = self.l2_processing_lock.lock().await;
+
+        let start_time = Instant::now();
+        info!("Starting database backup process...");
+
+        let timestamp = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
+        info!(
+            "Creating {} backup at path {}",
+            self.node_kind,
+            backup_path.display()
+        );
+
+        let mut handles = Vec::new();
+
+        {
+            let dbs = self.databases.read().unwrap();
+            for dir in &self.config.backup_dirs {
+                let path = backup_path.join(dir);
+                let db = dbs.get(dir).context("Missing required db")?.clone();
+                handles.push(tokio::task::spawn_blocking(move || db.create_backup(&path)));
+            }
+        }
+
+        // Wait for all dbs to starting backing up under lock before releasing
+        drop(l1_lock);
+        drop(l2_lock);
+
+        for handle in handles {
+            handle.await??;
+        }
```

Comment:
Given the above comment, i do believe we should await on all backup tasks to finish BEFORE we drop / release the locks. Releasing the locks and then awaiting on the tasks will trigger block production to resume and DB will be updated while it's being copied.